src/fsharp/service/ServiceLexing.fsi

```diff
@@ -256,3 +258,239 @@ module Keywords =
+        | Minus
+        | Dollar
+        | BarRightBracket
+        | BarRightBrace
+        | Underscore
+        | Semicolon
+        | SemicolonSemicolon
+        | LeftArrow
+        | Equals
+        | LeftBracket
+        | LeftBracketBar
+        | LeftBraceBar
+        | LeftBracketLess
+        | LeftBrace
+        | QuestionMark
+        | QuestionMarkQuestionMark
+        | Dot
+        | Colon
+        | ColonColon
+        | ColonGreater
+        | ColonQuestionMark
+        | ColonQuestionMarkGreater
+        | ColonEquals
+        | When
+        | While
+        | With
+        | Hash
+        | Ampersand
+        | AmpersandAmpersand
+        | Quote
+        | LeftParenthesis
+        | RightParenthesis
+        | Star
+        | Comma
+        | RightArrow
+        | GreaterBarRightBracket
+        | LeftParenthesisStarRightParenthesis
+        | Open
+        | Or
+        | Rec
+        | Then
+        | To
+        | True
+        | Try
+        | Type
+        | Val
+        | Inline
+        | Interface
+        | Instance
+        | Const
+        | Lazy
+        | OffsideLazy
+        | Match
+        | MatchBang
+        | Mutable
+        | New
+        | Of
+        | Exception
+        | False
+        | For
+        | Fun
+        | Function
+        | If
+        | In
+        | JoinIn
+        | Finally
+        | DoBang
+        | And
+        | As
+        | Assert
+        | OffsideAssert
+        | Begin
+        | Do
+        | Done
+        | DownTo
+        | Else
+        | Elif
+        | End
+        | DotDot
+        | DotDotHat
+        | BarBar
+        | Upcast
+        | Downcast
+        | Null
+        | Reserved
+        | Module
+        | Namespace
+        | Delegate
+        | Constraint
+        | Base
+        | LeftQuote
+        | RightQuote
+        | RightQuoteDot
+        | PercentOperator
+        | Binder
+        | Less
+        | Greater
+        | Let
+        | Yield
+        | YieldBang
+        | BigNumber
+        | Decimal
+        | Char
+        | Ieee64
+        | Ieee32
+        | NativeInt
+        | UNativeInt
+        | UInt64
+        | UInt32
+        | UInt16
+        | UInt8
+        | Int64
+        | Int32
+        | Int32DotDot
+        | Int16
+        | Int8
+        | FunkyOperatorName
+        | AdjacentPrefixOperator
+        | PlusMinusOperator
+        | InfixAmpersandOperator
+        | InfixStarDivideModuloOperator
+        | PrefixOperator
+        | InfixBarOperator
+        | InfixAtHatOperator
+        | InfixCompareOperator
+        | InfixStarStarOperator
+        | Identifier
+        | KeywordString
+        | String
+        | ByteArray
+        | Asr
+        | InfixAsr
+        | InfixLand
+        | InfixLor
+        | InfixLsl
+        | InfixLsr
+        | InfixLxor
+        | InfixMod
+
+    [<Struct;NoComparison;NoEquality;Experimental("This FCS API is experimental and subject to change.")>]
+    type public FSharpSyntaxToken =
```

Comment:
Note sure this should be a struct, it's getting pretty big (range is a big struct already, not sure about Parser.token).  Might be good to at least add an assert about it's expected `sizeof`?

Reply:
(BTW it would be good to get a modern measure on an appropriate size for structs in 32-bit and 64-bit processes, under typical copy/store loads)

Reply:
This will be a 24-byte sized struct in 64-bit and 20-byte sized struct in 32-bit. The rule of thumb max size for a struct is 16, so this is for sure over the limit and typically I stray away from it.

However, this API doesn't get invoked for every file when a project builds. It's simply an API to get token information on files we occasionally ask for, such as the token's range and kind, but in some cases a lot when we discover locations during find all refs in VS. If we make this a class, then we will allocate many amounts of these objects and puts pressure on the GC.

What we could do is remove `range` from `FSharpSyntaxToken`, and just pass the `range` as part of the callback, making `tokenCallback: (FSharpSyntaxToken -> range -> unit)`.

Another alternative would be to figure out if we could simply have two integers, representing start and ending positions based on the source's text, rather than the full information. That would put `FSharpSyntaxToken` at a perfect 16-byte struct on 64-bit. But this could be a lot of work to accomplish.

Reply:
It's OK, it's no big deal. (The 16byte guidance is dodgy anyway, dating ftom the data of pervasive 32bit, and not backed by any data AFAIK)